services/housekeeper/housekeeper.go

```diff
@@ -164,13 +225,46 @@ func (hk *Housekeeper) updateKnownValidators() {
 
 	for _, v := range validators {
 		pubkey := types.PubkeyHex(v.Validator.Pubkey)
-		err = hk.redis.SetKnownValidator(pubkey, v.Index)
+		err := hk.redis.SetKnownValidator(pubkey, v.Index)
 		if err != nil {
 			log.WithError(err).WithField("pubkey", pubkey).Fatal("failed to set known validator in Redis")
 		}
 	}
 }
 
+func (hk *Housekeeper) fetchValidators() map[types.PubkeyHex]beaconclient.ValidatorResponseEntry {
+	// return the first successful beacon node response
+	var result map[types.PubkeyHex]beaconclient.ValidatorResponseEntry
+	clients := hk.getBeaconClientsByLastResponse()
+
+	var mu sync.Mutex
+	for i, client := range clients {
+		log := hk.log.WithField("uri", client.GetURI())
+		log.Debug("fetching validators")
+
+		mu.Lock()
+		headSlot := hk.headSlot
+		mu.Unlock()
+
+		validators, err := client.FetchValidators(headSlot)
+		if err != nil {
+			hk.log.WithError(err).Error("failed to fetch validators")
+			continue
+		}
+
+		// Received successful response. Set this index as last successful beacon node
+		result = validators
```

Comment:
you could return result here, after setting lastHealthlyBeaconNodeIndex. Makes the code a bit simpler to understand